src/components/deployment/entity.ts

```diff
@@ -0,0 +1,128 @@
+import { Client } from 'pg';
+import { v4 as uuid } from 'uuid';
+
+import { IDeployment, IEntity, IQuery } from './types';
+
+const deploymentsTable = 'deployments';
+
+interface IBuiltQuery {
+  readonly query: ReadonlyArray<string>;
+  readonly values: ReadonlyArray<any>;
+}
+
+interface IBuiltQueryParams {
+  readonly key: string;
+  readonly operator?: string;
+  readonly value: any;
+}
+
+function where(startAt: number, ...filters: ReadonlyArray<IBuiltQueryParams>): IBuiltQuery {
+  return filters.reduce((all: IBuiltQuery, next: IBuiltQueryParams) => {
+    if (next.value !== undefined) {
+      const index = `$${startAt + all.values.length}`;
+      return {
+        query: [
+          ...all.query,
+          `"${next.key}" ${next.operator || '='} ${next.value === null ? 'NULL' : index}`,
+        ],
+        values: next.value === null ? all.values : [...all.values, next.value],
+      };
+    }
+
+    return all;
+  }, {query: [], values: []});
+}
+
+export async function countDeployments(client: Client, query: IQuery): Promise<number> {
+  const wheres = where(1,
+    {key: 'deletedAt', operator: 'IS', value: null},
+    {key: 'organizationGUID', value: query.organization},
+    {key: 'spaceGUID', value: query.space},
+  );
+
+  const result = await client.query({
+    name: 'count-deployments',
+    text: `SELECT COUNT(*)::integer as total FROM ${deploymentsTable} WHERE ${wheres.query.join(' AND ')}`,
+    values: [...wheres.values],
+  });
+
+  return result.rows[0].total;
+}
+
+export async function listDeployments(client: Client, query: IQuery): Promise<ReadonlyArray<IEntity>> {
+  const wheres = where(3,
+    {key: 'deletedAt', operator: 'IS', value: null},
+    {key: 'organizationGUID', value: query.organization},
+    {key: 'spaceGUID', value: query.space},
+  );
+
+  const result = await client.query({
+    name: 'list-deployments',
+    text: `SELECT * FROM ${deploymentsTable} WHERE ${wheres.query.join(' AND ')} LIMIT $1 OFFSET $2`,
```

Comment:
```suggestion
    text: `SELECT <specify field names instead of using *> FROM ${deploymentsTable} WHERE ${wheres.query.join(' AND ')} LIMIT $1 OFFSET $2`,
```